src/Features/Core/Portable/Completion/CompletionProvider.cs

```diff
@@ -34,13 +35,24 @@ protected CompletionProvider()
         public virtual bool ShouldTriggerCompletion(SourceText text, int caretPosition, CompletionTrigger trigger, OptionSet options)
             => false;
 
+        /// <summary>
+        /// Returns true if the character recently inserted or deleted in the text should trigger completion.
+        /// </summary>
+        /// <param name="languageServices">The language services available on the text document.</param>
+        /// <param name="text">The text that completion is occurring within.</param>
+        /// <param name="caretPosition">The position of the caret after the triggering action.</param>
+        /// <param name="trigger">The triggering action.</param>
+        /// <param name="options">The set of options in effect.</param>
+        internal virtual bool ShouldTriggerCompletion(HostLanguageServices languageServices, SourceText text, int caretPosition, CompletionTrigger trigger, OptionSet options)
+            => ShouldTriggerCompletion(text, caretPosition, trigger, options);
```

Comment:
not certain i get why we need this method just yet.

Reply:
see PR description above ^
We don't want to roundtrip document -> source text -> document, so we need to provide a way for the embedded languages to get the language service without round tripping

Reply:
this should be made more clear now